### PR TITLE
Doc bar snapping

### PR DESCRIPTION
**Pull request by nsobadjiev**
# Introduction

This is regarding the ticket about making some Document Bar improvements. More specifically, the snapping - every time the scroll view stops moving, it snaps to the closes cell.

Also, fixed a bug where a page in the document bar is still selected, even though the camera view is shown.

# How to test

Play around with the document bar

# Merge info

I'll do it manually

### CODE REVIEW

File: GiniSwitchSDK/GiniSwitchSDK/Classes/ExtractionsManager.swift [hunk 1]
```diff
@@ -258,7 +288,12 @@ class ExtractionsManager {
         status?.pages.forEach({ (page) in
             if let pageRef = page.href,
                 let scannedPage = scannedPages.page(for: pageRef) {
-                if scannedPage.status != page.pageStatus {
+                // ignore statue changes for pages scheduled for deletion and
+                // replacement. They might be marked as analysed, but since they
+                // will be replaced soon, the status shouldn't be displayed
+                if scannedPage.status != page.pageStatus &&
+                    scannedPage.status != .replaced &&
+                    scannedPage.status != .deleted {
```
Comment (EarlOfEgo) [inline GiniSwitchSDK/GiniSwitchSDK/Classes/ExtractionsManager.swift:296]: See https://github.com/gini/gini-switch-sdk-ios/pull/81#discussion_r135290856

File: GiniSwitchSDK/GiniSwitchSDK/Classes/PagesCollectionViewController.swift [hunk 2]
```diff
@@ -180,4 +188,60 @@ extension PagesCollectionViewController {
...
+        // delegate methods and calling it directly interferes somehow with its innerworkings,
+        // so selecting/deselecting cell might not work properly.
+        DispatchQueue.main.async {
+            guard let collectionView = self.pagesCollection else {
+                return
+            }
+            let cells = collectionView.visibleCells
+            var minDistance = collectionView.contentSize.width
+            let closestCell = cells.reduce(nil, { (closestCell, currentCell) -> UICollectionViewCell? in
+
```
Comment (EarlOfEgo) [inline GiniSwitchSDK/GiniSwitchSDK/Classes/PagesCollectionViewController.swift]: I think this empty line is quite disturbing for reading the anonymous function.

**Review by EarlOfEgo — APPROVED** [review]
LGMT

File: GiniSwitchSDK/GiniSwitchSDK/Classes/PagesCollectionViewController.swift (hunk 2 above)
Comment (nsobadjiev) [reply GiniSwitchSDK/GiniSwitchSDK/Classes/PagesCollectionViewController.swift]: It's called a closure, @EarlOfEgo , look it up!
But, yeah, ok

**Pull request merged**